### PR TITLE
tests/core/uc20-recovery: move recover mode helpers to generic testslib script

### PR DESCRIPTION
**Pull request by anonymouse64**
This is so we can re-use the helpers from other tests, where we have recover
mode specific behavior we want to test. Specifically I want to re-use it in the UC20 
recovery snap-repair spread test I will propose shortly based on top of this PR.
Moving it here reduces noise for that PR.

Also drive-by update to simplify the tests/core/gadget-update-pc test as it doesn't
need to specify the systems filter, since it's already in the core suite.

### CODE REVIEW

File: tests/lib/uc20-recovery.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+fi
+EOF
+    fi
+
+    # XXX: is this a race between spread seeing REBOOT and machine rebooting?
+    REBOOT
+}
+
+transition_to_run_mode() {
+    if [ -n "$1" ]; then
```
Comment (bboozzoo) [inline tests/lib/uc20-recovery.sh:68]: same here

File: tests/lib/uc20-recovery.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash -ex
+
+transition_to_recover_mode(){
+    if [ -n "$1" ]; then
+        local label=$1
+        HAVE_LABEL=1
```
Comment (bboozzoo) [inline tests/lib/uc20-recovery.sh:6]: we shuould probably have
```
local have_label
```
somewhere earlier

**Review by bboozzoo — APPROVED** [review]

File: tests/lib/uc20-recovery.sh (hunk 2 above)
Comment (anonymouse64) [reply tests/lib/uc20-recovery.sh:6]: done

File: tests/lib/uc20-recovery.sh (hunk 1 above)
Comment (anonymouse64) [reply tests/lib/uc20-recovery.sh]: done

**Review by mvo5 — APPROVED** [review]
Thank you

**Pull request merged**